### ISSUE TITLE
sql bug with  printFieldListFrom hook

### ISSUE THREAD

**Issue by atm-jpb** [Dolibarr/dolibarr#18594]
In many dolibarr lists, some tables are joined in the WHERE clause and not with a join statement  in FROM clause. this  can cause execution bugs if we use the hook 'printFieldListFrom'

example :

$sql .= "SELECT * FROM A AS a";
$sql .= " LEFT JOIN  B as b on b.field = a.field";
$sql .= ",C as c";

(C and E will be joined to A on where clause after.)

then if we use the hook printFieldListFrom to complete sql list like  this : 
 $hookmanager->resPrint = " LEFT JOIN F as f On f.field = a.field ";  

we get a execution bug cause alias a not recognized by mysql.

how to prevent it ? 

i think we should always use Join statements in FROM clause  instead of 
join fields in where clause, like this  : 
$sql .= " LEFT JOIN  C as c on c.field = a.field";
instead of :
$sql .= ",C as c";

would you agree if i replace in all sql lists,  all joins done in where clause by JOIN statement in FROM clause ? 
it will handle this case.

**Comment by atm-maxime** [comment]
@eldy question about SQL requests table joins, can you explain why we are using sometimes "JOIN" and sometimes "," ? And how should we deal with the hook ?

**Comment by atm-jpb** [comment]
@eldy

**Comment by atm-maxime** [comment]
@eldy ?

**Comment by eldy** [comment]
Yes switching on JOIN instead of coma separated tables is a good solution to be able to use hook to add tables whatever is the order of tables. You can start in this direction. First on few exemples to definitely validate this but i see no side effect for the moment...

**Comment by atm-maxime** [comment]
@atm-jpb 😉
